src/extensions/default/Inline3DParametersEditor/ParameterEditor.js

```diff
@@ -0,0 +1,149 @@
+        this.$zValueSlider = this.$element.find("#_z");
+        this._x = parseFloat(this.$xValueSlider.val());
+        this._y = parseFloat(this.$yValueSlider.val());
+        this._z = parseFloat(this.$zValueSlider.val());
+
+        this.$selectionBase = this.$element.find(".selector-base");
+
+        // Attach event listeners to main UI elements
+        this._addListeners();
+    }
+
+
+    /** Returns the root DOM node of the UI */
+    ParameterEditor.prototype.getRootElement = function () {
+        return this.$element;
+    };
+
+    ParameterEditor.prototype._addListeners = function () {
+        var self = this;
+        this.$xValueSlider.mousedown(function(e) {
+            self._x = parseFloat(e.currentTarget.value);
+            self._position = e.clientX;
+        });
+
+        this.$yValueSlider.mousedown(function(e) {
+            self._y = parseFloat(e.currentTarget.value);
+            self._position = e.clientX;
+        });
+
+        this.$zValueSlider.mousedown(function(e) {
+            self._z = parseFloat(e.currentTarget.value);
+            self._position = e.clientX;
+        });
+
+        
+        this.$xValueSlider.change(function(e) {
+            self._commitParameters(self.$xValueSlider.val() + " 3 0");
+        });
+
+        this._registerDragHandler(this.$xValueSlider, this._handleXValueSliderDrag);
+        this._registerDragHandler(this.$yValueSlider, this._handleYValueSliderDrag);
+        this._registerDragHandler(this.$zValueSlider, this._handleZValueSliderDrag);
+
+    };
+
+    ParameterEditor.prototype.focus = function () {
+        if (!this.$selectionBase.is(":focus")) {
+            this.$selectionBase.focus();
+            return true;
+        }
+        return false;
+    };
+
+    ParameterEditor.prototype._commitParameters = function (parameters) {
+        this._callback(parameters);
+    };
+
+    ParameterEditor.prototype.setParametersFromString = function (parameters) {
+        this.setSliderProperties(parameters);
+        this._commitParameters(parameters, true);
+    };
+
+    ParameterEditor.prototype.isValidSetOfParameters = function (parameters) {
+        var ParameterRegex = new RegExp(/((-)?\d+(\.\d+)?) ((-)?\d+(\.\d+)?) ((-)?\d+(\.\d+)?)/g);
+        return ParameterRegex.test(parameters);
+    };
+
+    ParameterEditor.prototype.setSliderProperties = function (parameters) {
+        var parametersArray = parameters.split(" ");
+        this.$xValueSlider.val(parametersArray[0]);
+        this.$yValueSlider.val(parametersArray[1]);
+        this.$zValueSlider.val(parametersArray[2]);        
+    };
+
+    function _getNewOffset(pos, zeroPos) {
+        var offset = pos - zeroPos;
+        return offset/MODIFIER;
+    };
+
+    ParameterEditor.prototype._handleXValueSliderDrag = function(event) {
+        var xPos = this._position;
+        var offset = _getNewOffset(event.clientX, xPos );
+        var n = this._x + offset;
+        this.$xValueSlider.val(n.toFixed(2));
```

Comment:
I'm always wary of `toFixed()`.  Are you really needing a string with 2 places padded vs. just a number?